### PR TITLE
CORTX-29404 Provisioner changes to support partition stob

### PR DESCRIPTION
**Pull request by zahidsk**
Signed-off-by: Zahid Shaikh <zahid.shaikh@seagate.com>

# Problem Statement
- Provisioner changes to support partition stob

# Design
-  [For Feature, Post the link for design](https://jts.seagate.com/browse/CORTX-27677)

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: src/provisioner/cluster.py [hunk 1]
```diff
@@ -110,16 +111,9 @@ def _get_storage_kv_list(self, storage_spec: dict, node_id: str):
         """"Return list of confstore key-values for storage."""
         # Create confstore keys,from storage_spec(config from config.yaml).
         storage_kv_list = []
-        key_prefix = f'node>{node_id}>storage'
-        # TODO: Need to remove num cvg_count key later
-        storage_kv_list.append((f'{key_prefix}>cvg_count', len(storage_spec)))
```
Comment (tanujashinde0405) [inline src/provisioner/cluster.py:115]: if we remove cvg_count as part of this PR, then the flow will break right? as lakshita has created two bugs for motr and hare team.

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

File: src/provisioner/cluster.py (hunk 1 above)
Comment (tanujashinde0405) [reply src/provisioner/cluster.py:115]: Let's not remove it as part of this PR, let's wait for motr and hare fixed then we will remove cvg_count. for now, let's only focus on removing metadata dependency from provisioner code.

File: src/provisioner/cluster.py [hunk 2]
```diff
@@ -75,7 +75,8 @@ def add_node(self, node: dict):
         self._validate(node)
         self._node_list.append(node)
 
-    def _get_kvs(self, prefix, node):
+    def _get_kvs(self, prefix: str, node: dict):
+        """ Converts dict into list of keys and values """
```
Comment (tanujashinde0405) [inline src/provisioner/cluster.py:79]: fix codacy.

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

File: src/provisioner/cluster.py (hunk 2 above)
Comment (zahidsk) [reply src/provisioner/cluster.py]: Fixed

File: src/provisioner/cluster.py (hunk 1 above)
Comment (zahidsk) [reply src/provisioner/cluster.py:115]: Removed

**Review by tanujashinde0405 — APPROVED** [review]

**Comment by zahidsk** [comment]
retest this please

**Comment by zahidsk** [comment]
Infra issue for 1N K8s Deployment : associated vm not reachable and having authentication issue
triggered build manually with my vm:
https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/K8s-1N-deployment/557/
It is **successful**

**Review by kailashyadav11 — APPROVED** [review]

File: src/provisioner/cluster.py [hunk 3]
```diff
@@ -110,16 +111,11 @@ def _get_storage_kv_list(self, storage_spec: dict, node_id: str):
         """"Return list of confstore key-values for storage."""
         # Create confstore keys,from storage_spec(config from config.yaml).
         storage_kv_list = []
-        key_prefix = f'node>{node_id}>storage'
+        key_prefix = f'node>{node_id}'
```
[the remark this answers is not in the document]
Comment (zahidsk) [reply src/provisioner/cluster.py]: reverted the storage key change

**Pull request merged**